content/copilot/index.md

```diff
@@ -0,0 +1,158 @@
+---
+title: Docker for GitHub Copilot
+description: Learn how to use the Docker Extension for GitHub Copilot to streamline Docker-related tasks directly within GitHub and Visual Studio Code.
+keywords: Docker, GitHub Copilot, extension, Visual Studio Code, chat, ai, containerization
+sitemap: false
+---
+
+> **Early Access**
+>
+> The Docker for GitHub Copilot extension is in [Early Access](../../release-lifecycle.md).
+> Only members of the early access program can access this feature.
+> To request access, [apply to the Early Access Program here](https://www.docker.com/docker-desktop-preview-program/).
+{ .restricted }
+
+The Docker Extension for GitHub Copilot integrates Docker's capabilities with
+GitHub Copilot, providing assistance with containerizing applications and
+generating Docker assets. This extension helps you streamline Docker-related
+tasks directly within GitHub and Visual Studio Code.
+
+The key features are:
+
+- Ask questions and receive responses about containerization directly within
+  GitHub or Visual Studio Code.
+- Automatically generate Dockerfiles, Docker Compose files, and .dockerignore
+  files for a project, using [`docker init`](../reference/cli/docker/init.md).
+- Open pull requests with generated Docker assets directly from the chat
+  interface.
+- Get summaries of project vulnerabilities from Docker Scout and receive next
+  steps via CLI.
+
+## Supported languages
+
+Currently, the Docker Extension for GitHub Copilot supports the following
+programming languages:
+
+- Dockerfile
+- Java
+- JavaScript
+- Python
+- TypeScript
+
+## Get Started
+
+Here's how to get started with the Docker Extension for GitHub Copilot.
+
+### Prerequisites
+
+Before you start, ensure that:
+
+- You have been granted access to the private beta program.
```

Comment:
We may want to link to https://github.blog/2024-05-21-introducing-github-copilot-extensions/? This has details that explain that it's currently a closed beta etc.